File: 0007-reverse-integer/0007-reverse-integer.java
class Solution {
    public int reverse(int x) {
        int num = x;
        int result = 0;
        if (x < 0) {
            x = 0 - x;
        }
        
        while (x > 0) {
            if(900000000 < result) {
                return 0;
            }
            
            result = (result * 10) + (x % 10);
            x /= 10;
            if (result < 0) {
                return 0;
            }
        }
         
        if (num < 0) {
            return result * (-1);
        }
       
        return result;
    }
}

File: 0007-reverse-integer/0007-reverse-integer.go
func reverse(x int) int {
    num := x
    result := 0
    if x < 0 {
        x = x * -1
    }
    
    for x > 0 {
        if result > 700000000 {
            return 0;
        }
        
        result = (result * 10) + (x % 10)
        x = x / 10
        
        if result < 0 {
            return 0
        }
    }
    
    if result > 2147483650 {
        return 0
    }
    
    if num < 0 {
        return -result
    }
    
    return result
}